caterva2/services/sub.py

```diff
@@ -358,75 +356,42 @@ async def download_data(path: str, slice_: str = None, download: bool = False):
     slice_ : str
         The slice to fetch.
     download : bool
-        Whether to download the dataset in the downloads/ dir.  If False, the data is
```

Comment:
The changes look good to me in general, however I find this API endpoint of the subscriber quite confusing, as I gather that it's merging three different functionalities together for the sake of sharing the code path that gets the file to the subscriber:

- An operation that streams the Blosc2 dataset back to the client and doesn't allow slices (`download=False`, no slice). No Content-Type in response.
- An operation that returns a URL where the client can get the whole Blosc2 dataset (`download=True`). JSON Content-Type in response.
- An operation that returns a NumPy array to the client (`download=False`, with slice). No Content-Type in response.

It really took me a while to understand this, but it looks like this PR just inherits the previous behaviour, so you may proceed to merge it, and this may be fixed later.

Reply:
Agreed.  Let's merge this now, and it would be great if you want to do some refactoring later on.